### PR TITLE
Support fetching imports from a Module

### PR DESCRIPTION
**Pull request by davejcameron**
Add support for fetching the list of Module imports.

### CODE REVIEW

File: ext/src/ruby_api/module.rs [hunk 1]
```diff
@@ -106,6 +109,24 @@ impl Module {
...
+    /// @yard
+    /// Returns the list of imports that this Module has and must be satisfied.
+    /// @return [Array<Hash>] An array of hashes containing import information
+    pub fn imports(&self) -> Result<RArray, Error> {
+        let module = self.get();
+        let imports = module.imports();
+
+        let result = RArray::new();
+        for import in imports {
+            let hash = RHash::new();
```
Comment (saulecabrera) [inline ext/src/ruby_api/module.rs:122]: The exports' length is known, I believe here we can use `RArray::with_capacity(imports.len())` to minimize the cost of reallocations. We can do the same with the usage of `RHash`.

File: ext/src/ruby_api/module.rs [hunk 2]
```diff
@@ -106,6 +109,24 @@ impl Module {
...
+            hash.aset("module", import.module())?;
+            hash.aset("name", import.name())?;
+            hash.aset("type", format!("{:?}", import.ty()))?;
```
Comment (saulecabrera) [inline ext/src/ruby_api/module.rs:125]: I'm not sure we want to return a string for the type, ideally we'd want a proper type representing the extern. I believe you can use https://github.com/bytecodealliance/wasmtime-rb/blob/main/ext/src/ruby_api/externals.rs#L110

File: ext/src/ruby_api/externals.rs [hunk 3]
```diff
@@ -1,13 +1,98 @@
...
+        }
+    }
+}
+unsafe impl Send for ExternType {}
+
+impl ExternType {
+    /// @yard
+    /// Returns the exported function's FuncType or raises a `{ConversionError}` when the export is not a
+    /// function.
+    /// @return [Func] The exported function.
```
Comment (saulecabrera) [inline ext/src/ruby_api/externals.rs:47]: I believe here it should be

```
@return [FuncType]
```

In the signatures below as well?

**Review by saulecabrera — COMMENTED** [review]
Looks good to me; there are couple of docs that need updating and after we should be able to land this one.

File: ext/src/ruby_api/externals.rs (hunk 3 above)
Comment (mkcny) [reply ext/src/ruby_api/externals.rs]: Good catch! Fixed.

**Pull request merged**